compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/ArgumentsGenerationUtils.kt

```diff
@@ -61,6 +61,34 @@ import kotlin.math.min
 fun StatementGenerator.generateReceiverOrNull(ktDefaultElement: KtElement, receiver: ReceiverValue?): IntermediateValue? =
     receiver?.let { generateReceiver(ktDefaultElement, receiver) }
 
+fun StatementGenerator.generateContextReceiverForDelegatingConstructorCall(
```

Comment:
I assume there are corresponding tests in some of the commits below, right?

Reply:
Yes, the history could be better. [constructors.kt](https://github.com/JetBrains/kotlin/blob/rr/shadrina/context-receivers/compiler/testData/codegen/box/extensionClasses/constructors.kt) checks this part.